server/client.go

```diff
@@ -588,37 +588,57 @@ func (c *client) subsAtLimit() bool {
 	return c.msubs != jwt.NoLimit && len(c.subs) >= int(c.msubs)
 }
 
+func minLimit(value *int32, limit int32) bool {
+	if *value != jwt.NoLimit {
+		if limit != jwt.NoLimit {
+			if limit < *value {
+				*value = limit
+				return true
+			}
+		}
+	} else if limit != jwt.NoLimit {
+		*value = limit
+		return true
+	}
+	return false
+}
+
 // Apply account limits
 // Lock is held on entry.
 // FIXME(dlc) - Should server be able to override here?
 func (c *client) applyAccountLimits() {
 	if c.acc == nil || (c.kind != CLIENT && c.kind != LEAF) {
 		return
 	}
-
-	// Set here, will need to fo checks for NoLimit.
-	if c.acc.msubs != jwt.NoLimit {
-		c.msubs = c.acc.msubs
-	}
-	if c.acc.mpay != jwt.NoLimit {
-		c.mpay = c.acc.mpay
+	c.mpay = jwt.NoLimit
+	c.msubs = jwt.NoLimit
+	if c.opts.JWT != "" { // user jwt implies account
+		if uc, _ := jwt.DecodeUserClaims(c.opts.JWT); uc != nil {
+			c.mpay = int32(uc.Limits.Payload)
+			c.msubs = int32(uc.Limits.Subs)
+		}
 	}
-
+	minLimit(&c.mpay, c.acc.mpay)
```

Comment:
Say `uc.Limits.Payload` is 1000, then c.may will be 1000. Now say that account's mpay is 500, I don't think that minLimit does the right thing? (assuming that we should limit to the account's limit, but maybe my assumption is wrong here).

Reply:
c.mpay will be used to compare and accumulate the lowest value.
Since c.mpay contains the value used for comparing, sending the async info will pick up whatever minLimit picks.